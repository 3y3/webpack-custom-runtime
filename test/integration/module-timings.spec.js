'use strict';

const cct = require('./utils/custom-configurable-test');
const WebpackCustomRuntime = require('../../').CustomRuntimePlugin;

describe('ModuleTimings', function() {

    function cit(name, options, config) {
        cct(name, Object.assign({
            plugins: [
                new WebpackCustomRuntime({ behavior: '' }),
                new WebpackCustomRuntime.ModuleTimings(options),
            ]
        }, config));
    }

    cit('should setup default timings', {});

    cit('should configure timing function', {
        timer: 'Date.now'
    });
});
