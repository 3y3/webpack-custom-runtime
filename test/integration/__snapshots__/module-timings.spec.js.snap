// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleTimings should configure timing function: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ Change 1 @@
-/******/ 	  return Promise.resolve();
+/******/ 	  return [
+/******/ 	    function(installedChunks, chunkId, result) {
+/******/ 	      var ns = __webpack_require__.ModuleTimings;
+/******/
+/******/ 	      ns[chunkId] = ns[chunkId] || {};
+/******/ 	      ns[chunkId].loadingEnd = Date.now();
+/******/
+/******/ 	      return result;
+/******/ 	    }
+/******/ 	  ].reduce(function(promise, strategy) {
+/******/ 	    return promise.then(function(result) {
+/******/ 	      return strategy(installedChunks, chunkId, result);
+/******/ 	    });
+/******/ 	  }, Promise.resolve());
@@ Change 2 @@
-/******/ 	  return Promise.resolve();
+/******/ 	  return [
+/******/ 	    function(installedChunks, chunkId, result) {
+/******/ 	      var ns = __webpack_require__.ModuleTimings;
+/******/
+/******/ 	      ns[chunkId] = ns[chunkId] || {};
+/******/ 	      ns[chunkId].loadingStart = Date.now();
+/******/
+/******/ 	      return result;
+/******/ 	    }
+/******/ 	  ].reduce(function(promise, strategy) {
+/******/ 	    return promise.then(function(result) {
+/******/ 	      return strategy(installedChunks, chunkId, result);
+/******/ 	    });
+/******/ 	  }, Promise.resolve(result));
@@ Change 3 @@
+/******/ 	__webpack_require__.ModuleTimings = __webpack_require__.ModuleTimings || {};
@@ Change 4 @@
+/******/
"
`;

exports[`ModuleTimings should setup default timings: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ Change 1 @@
-/******/ 	  return Promise.resolve();
+/******/ 	  return [
+/******/ 	    function(installedChunks, chunkId, result) {
+/******/ 	      var ns = __webpack_require__.ModuleTimings;
+/******/
+/******/ 	      ns[chunkId] = ns[chunkId] || {};
+/******/ 	      ns[chunkId].loadingEnd = performance.now();
+/******/
+/******/ 	      return result;
+/******/ 	    }
+/******/ 	  ].reduce(function(promise, strategy) {
+/******/ 	    return promise.then(function(result) {
+/******/ 	      return strategy(installedChunks, chunkId, result);
+/******/ 	    });
+/******/ 	  }, Promise.resolve());
@@ Change 2 @@
-/******/ 	  return Promise.resolve();
+/******/ 	  return [
+/******/ 	    function(installedChunks, chunkId, result) {
+/******/ 	      var ns = __webpack_require__.ModuleTimings;
+/******/
+/******/ 	      ns[chunkId] = ns[chunkId] || {};
+/******/ 	      ns[chunkId].loadingStart = performance.now();
+/******/
+/******/ 	      return result;
+/******/ 	    }
+/******/ 	  ].reduce(function(promise, strategy) {
+/******/ 	    return promise.then(function(result) {
+/******/ 	      return strategy(installedChunks, chunkId, result);
+/******/ 	    });
+/******/ 	  }, Promise.resolve(result));
@@ Change 3 @@
+/******/ 	__webpack_require__.ModuleTimings = __webpack_require__.ModuleTimings || {};
@@ Change 4 @@
+/******/
"
`;
